packages/plugin-ext/src/main/browser/view/plugin-view-widget.ts

```diff
@@ -96,10 +100,39 @@ export class PluginViewWidget extends Panel implements StatefulWidget {
         }
     }
 
+    private _message: string | undefined;
+    get message(): string | undefined {
+        return this._message;
+    }
+
+    set message(message: string | undefined) {
+        this._message = message;
+        this.updateWidgetMessage();
+    }
+
+    private updateWidgetMessage(): void {
+        const widget = this.widgets[0];
+        if (widget) {
+            if (widget instanceof TreeViewWidget) {
+                widget.message = this._message;
+            }
+        }
+    }
+
+    addWidget(widget: Widget): void {
```

Comment:
we should revert `addWidget` and `insertWidget` they are not for rendering.

Reply:
It is not about rendering. `addWidget` and `insertWidget` are overridden in order to safely set message value on the widget, because `$setMessage` API is invoked mostly when the widget creation is started, but not finished (so, the widget is not set into the View's widgets[] and, as such is not accessible).  
Otherwise, I don't know a way to properly realize `$setMessage` API other than `debouncing` which we were trying to get rid of. Any suggestions?

Reply:
Let's stick with it for now and wait for a bug. Such approach seems to work timely wise concretely for this extension. If all extension reveal the widget before setting message then it should work generally.